superbuild/CMakeLists.txt

```diff
@@ -64,9 +82,42 @@ if (NOT PREBUILT_STAGE2 STREQUAL "NO" AND NOT PREBUILT_STAGE2_PATH STREQUAL "NO"
 endif()
 
 set(STAGE2_PATH "${CMAKE_BINARY_DIR}/stage2")
+if (NOT PREBUILT_STAGE1_PATH STREQUAL "NO")
+    set(STAGE2_PATH "${PREBUILT_STAGE1_PATH}")
+endif()
 if (NOT PREBUILT_STAGE2_PATH STREQUAL "NO")
     set(STAGE2_PATH "${PREBUILT_STAGE2_PATH}")
 endif()
+set(STAGE3_PATH "${CMAKE_BINARY_DIR}/stage3")
+
+if (NOT PREBUILT_STAGE1_PATH STREQUAL "NO" AND NOT PREBUILT_STAGE2_PATH STREQUAL "NO")
+    message(FATAL_ERROR "PREBUILT_STAGE1_PATH and PREBUILT_STAGE2_PATH is incompatible together.")
+endif()
+
+if (BUILD_STAGE1_TOOLCHAIN_ONLY OR BUILD_STAGE2_TOOLCHAIN_ONLY OR
+    BUILD_SPIRV_TRANSLATOR_ONLY OR BUILD_VC_INTRINSICS_ONLY OR BUILD_L0_LOADER_ONLY)
+    set(XE_DEPS OFF)
+endif()
+
+if (BUILD_SPIRV_TRANSLATOR_ONLY OR BUILD_VC_INTRINSICS_ONLY OR BUILD_L0_LOADER_ONLY)
+    set(INSTALL_WITH_XE_DEPS ON)
+endif()
+
+if (APPLE)
+    # XE are not supported on macOS targets
+    set(XE_DEPS OFF)
+
+    if (CMAKE_SYSTEM_PROCESSOR MATCHES "(x86)|(X86)|(amd64)|(AMD64)")
+        set (MACOS_VERSION_MIN 10.12)
```

Comment:
I need to double check this logic. It matches `alloy.py`, but I think the intention was to have `10.12` vs `11.0` by default for `x86_64` vs `arm64`, but do `10.12` in case of universal binary. AR for me - to check this.

Reply:
Here's the logic for selecting macOS version target is here: https://github.com/ispc/ispc/blob/1c6074be25da008b6a5756868011b784b425de47/CMakeLists.txt#L38

I.e. if either:
- building non-universal (native) binaries on `x86_64`, or
- universal binaries on `arm64`
then macOS target version is `10.12`. Otherwise `11.0`.

Note that you need to check only `x86_64`, no need to check "(X86)|(amd64)|(AMD64)" - because Mac hardware doesn't have this variability.

Reply:
So, I fixed that as follow: only arm non-universal is `11.0`, all other variants is `10.12`.